internal/web/server.go

```diff
@@ -0,0 +1,179 @@
+package web
+
+import (
+	"context"
+	"embed"
+	"fmt"
+	"html/template"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/styrainc/regal/internal/explorer"
+	"github.com/styrainc/regal/internal/lsp/cache"
+	"github.com/styrainc/regal/internal/lsp/clients"
+)
+
+//go:embed assets/*
+var assets embed.FS
+
+//nolint:gochecknoglobals,lll
+var tpl = template.Must(template.New("main").Parse(`
+<!DOCTYPE html>
+<html>
+	<head>
+		<meta charset="utf-8"/>
+		<link rel="stylesheet" href="/assets/missing.min.css">
+		<script src="https://unpkg.com/htmx.org@1.8.4" integrity="sha384-wg5Y/JwF7VxGk4zLsJEcAojRtlVp1FKKdGy1qN+OMtdq72WRvX/EdRdqg/LOhYeV" crossorigin="anonymous"></script>
+	<body>
+		<header>
+			<h1>OPA Explorer<v-h>:</v-h><sub-title>Inspect compiler stages</sub-title></h1>
+			<nav>
+				<p class="tool-bar">
+					<img src="https://openpolicyagent.org/badge/v0.68.0"/>
+					<a href="https://github.com/srenatus/opa-explorer">GitHub</a>
+				</p>
+			</nav>
+		</header>
+		<main class="crowded">
+			<form>
+				<div class="f-row">
+					<textarea name="code"
+					  style="height: 300px"
+					  hx-post="/?tmpl=output"
+					  hx-target="#output"
+					  hx-trigger="keyup changed delay:200ms"
+					  class="flex-grow:1 monospace"
+					>{{ .Code }}</textarea>
+				</div>
+			</form>
+			<section id="output">
+				{{ block "output" . }}
+				{{ range .Result }}
+				<details class={{ .Class }} {{ if .Show }}open{{ end }}>
+					<summary>{{ .Stage }}</summary>
+					<pre><code>{{ .Output }}</code></pre>
+				</details>
+				{{ end }}
+				{{ end }}
+			</section>
+		</main>
+	</body>
+</html>
+`))
+
+type Server struct {
+	cache        *cache.Cache
+	workspaceURI string
+	client       clients.Identifier
+	baseURL      string
+}
+
+func NewServer(cache *cache.Cache) *Server {
+	return &Server{cache: cache}
+}
+
+func (s *Server) SetWorkspaceURI(uri string) {
+	s.workspaceURI = uri
+}
+
+func (s *Server) SetClient(client clients.Identifier) {
+	s.client = client
+}
+
+func (s *Server) GetBaseURL() string {
+	return s.baseURL
+}
+
+type state struct {
+	Code   string
+	Result []stringResult
+}
+
+type stringResult struct {
+	Show   bool
+	Class  string
+	Stage  string
+	Output string
+}
+
+func (s *Server) Start(_ context.Context) {
```

Comment:
```suggestion
func (s *Server) Start() {
```
If we don't use the context, let's remove it.